aicsimageio/transforms.py

```diff
@@ -105,59 +55,116 @@ def reshape_data(
+                # first and last slices of an image [0, -1]
+                if isinstance(dim_operator, list):
+                    check_selection_max = max(dim_operator)
+
+                # No need to check slices as slices work with array syntax and will
+                # ignore out-of-bounds indicies
+                if isinstance(dim_operator, slice):
+                    check_selection_max = 0
+            else:
+                # Nothing was requested from this dimension
+                dim_operator = slice(None, None, None)
+
+                # No op means that it doesn't matter how much data is in this dimension
+                check_selection_max = 0
+
+        # Not in given dims means that it is a fixed integer selection
+        else:
+            if dim in kwargs:
+                # Integer requested
+                dim_operator = kwargs.get(dim)
+
+                # Check that integer
+                check_selection_max = dim_operator
+            else:
+                # Dimension wasn't included in kwargs, default to zero
                 log.warning(
-                    f"Data has dimension {dim} with depth {data.shape[index]}, "
+                    f"Data has dimension {dim} with depth {data.shape[dim_index]}, "
                     f"assuming {dim}=0 is the desired value, "
                     f"if not the case specify {dim}=x where "
-                    f"x is an integer in [0, {data.shape[index]})."
-                )
-                index_depth = 0
-            if index_depth >= data.shape[index]:
-                raise IndexError(
-                    f"Dimension specified with {dim}={index_depth} "
-                    f"but Dimension shape is {data.shape[index]}."
+                    f"x is an integer in [0, {data.shape[dim_index]}])."
                 )
-            planes = _split(
-                operator, data, data.shape[index], axis=index
-            )  # split dim into list of arrays
-            data = planes[index_depth]  # take the specified value of the dim
-        data = operator.squeeze(data, axis=index)  # remove the dim from ndarray
-        new_dims = (
-            new_dims[0:index] + new_dims[index + 1 :]
-        )  # clip out the Dimension from new_dims
+                dim_operator = 0
+                check_selection_max = 0
+
+            # Remove dim from new dims as it is fixed size
+            new_dims = new_dims.replace(dim, "")
+
+        # Check that fixed integer request isn't outside of request
+        if check_selection_max > data.shape[dim_index]:
+            raise IndexError(
+                f"Dimension specified with {dim}={dim_operator} "
+                f"but Dimension shape is {data.shape[dim_index]}."
+            )
+
+        # All checks and operations passed, append dim operation to getitem ops
+        getitem_ops.append(dim_operator)
+
+    # Run getitems
+    data = data[tuple(getitem_ops)]
```

Comment:
I like this, nice and clean!  I might consider renaming getitem_ops to dimension_specs, and dim_operator to dimension_spec or something like that.  I find the variable names a little cryptic.